dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/internal/DLCTransactionProcessing.scala

```diff
@@ -0,0 +1,275 @@
+package org.bitcoins.dlc.wallet.internal
+
+import org.bitcoins.core.api.wallet.db.SpendingInfoDb
+import org.bitcoins.core.protocol.dlc.models.DLCMessage._
+import org.bitcoins.core.protocol.dlc.models._
+import org.bitcoins.core.protocol.script._
+import org.bitcoins.core.protocol.tlv._
+import org.bitcoins.core.protocol.transaction.{Transaction, WitnessTransaction}
+import org.bitcoins.core.util.FutureUtil
+import org.bitcoins.core.wallet.utxo.AddressTag
+import org.bitcoins.crypto.DoubleSha256DigestBE
+import org.bitcoins.dlc.wallet.DLCWallet
+import org.bitcoins.dlc.wallet.models._
+import org.bitcoins.wallet.internal.TransactionProcessing
+
+import scala.concurrent._
+import scala.util.Try
+
+/** Overrides TransactionProcessing from Wallet to add extra logic to
+  * process transactions that could from our own DLC.
+  */
+private[bitcoins] trait DLCTransactionProcessing extends TransactionProcessing {
```

Comment:
Yikes, in general we need to stop using inheritance in the wallet and start using `has-a` rather than `is-a`. This isn't going to get changed in this PR, but needs to be taken care of eventually.